### ISSUE TITLE
No persistence with Linux Mint 21.3 Cinnamon 64-bit

### ISSUE THREAD

**Issue by BenthamITServices** [pbatard/rufus#2428]
﻿

Checklist
---------
- [x] I looked at https://github.com/pbatard/rufus/wiki/FAQ to see if my question has already been answered.
- [x] I performed a search in the issue tracker for similar issues using keywords relevant to my problem, such as the error message I got from the log.
- [x] I clicked the 'Log' button (🗒️) or pressed <kbd>Ctrl</kbd>-<kbd>L</kbd> in Rufus, or used [DebugView](https://learn.microsoft.com/en-us/sysinternals/downloads/debugview), and copy/pasted the log into the section that says `<FULL LOG>` below.
- [x] The log I am copying is the FULL log, starting with the line `Rufus version: x.y.z` - I have NOT removed any part of it.

Additionally (if applicable):
- [ ] I ran a bad blocks check, by clicking _Show advanced format options_ then _Check device for bad blocks_, and confirmed that my USB is not defective.
- [x] I also tried one or more of the following:
  - [x] Using a different USB drive.
  - [x] Plugging the USB into a different port.
  - [ ] Running Rufus on a different computer.
- [ ] If using an image, I clicked on the `(✓)` button to compute the MD5, SHA1 and SHA256 checksums, which are therefore present in the log I copied. I confirmed, by performing an internet search, that these values match the ones from the official image.

Issue description
-----------------
I can not get persistence to work for Linux Mint 21.3 Cinnamon 64-bit using Rufus, but I can create a persistent installation using Ventoy and a pre-created persistence file.  I can create persistent USB drives for Debian or Ubuntu using Rufus.

[rufus.log](https://github.com/pbatard/rufus/files/14193026/rufus.log)

**Comment by pbatard** [comment]
Normally because there are too many Linux distros all using different ways of enabling persistence (they can't even agree on whether to use `persistence` or `persistent` as the kernel option), I would just point you to the fact that Rufus is only designed to support Debian or Ubuntu persistence and that [_"Other distros can go to hell"_](https://github.com/pbatard/rufus/blob/0f23c4718421c8bd7dd4810b7ef759267d535f83/src/iso.c#L390).

However, luckily for you, over on the Mint forums, someone successfully lied to me that, because Mint was a direct Ubuntu derivative, persistence used to work for older versions of the distro, and that it had simply been broken with newer releases (because they force Rufus to use NTFS rather than FAT32 do to Mint relying on the use of symbolic links), which led me to a painful exercise of refactoring the partition creation code just so we could accommodate persistence with NTFS. What this means is that so I was already in the process of enabling persistence for Mint when you opened the issue.

Long story short, persistence should work fine with Mint with the next release or Rufus. However, because of the amount of extra work trying to support this specific distro has forced me to undertake over the last 6-12 months, don't expect much when the Mint developers inevitably decide to deviate from Ubuntu further, and break persistence ~~yet again~~.
